app/models/journal/aggregated_journal.rb

```diff
@@ -0,0 +1,168 @@
+    # its predecessor and successor.
+    def sql_rough_group(uid)
+      "SELECT predecessor.*, #{sql_group_counter(uid)} AS group_number
+      FROM #{sql_rough_group_from_clause(uid)}
+      LEFT OUTER JOIN journals successor
+        ON predecessor.version + 1 = successor.version AND
+           predecessor.journable_type = successor.journable_type AND
+           predecessor.journable_id = successor.journable_id
+      WHERE predecessor.user_id != successor.user_id OR
+            (predecessor.notes != '' AND predecessor.notes IS NOT NULL) OR
+            #{sql_beyond_aggregation_time?('predecessor', 'successor')} OR
+            successor.id IS NULL"
+    end
+
+    # The "group_number" required in :sql_rough_group has to be generated differently depending on
+    # the DBMS used. This method returns the apropriate statement to be used inside a select to
+    # obtain the current group number.
+    # The :uid parameter allows to define non-conflicting variable names (for MySQL).
+    def sql_group_counter(uid)
+      if OpenProject::Database.mysql?
+        group_counter = mysql_group_count_variable(uid)
+        "(#{group_counter} := #{group_counter} + 1)"
+      else
+        'row_number() OVER ()'
+      end
+    end
+
+    # MySQL requires some initialization to be performed before being able to count the groups.
+    # This method allows to inject further FROM sources to achieve that in a single SQL statement.
+    # Sadly MySQL requires the whole statement to be wrapped in parenthesis, while PostreSQL
+    # prohibits that.
+    def sql_rough_group_from_clause(uid)
+      if OpenProject::Database.mysql?
+        "(journals predecessor, (SELECT #{mysql_group_count_variable(uid)}:=0) number_initializer)"
+      else
+        'journals predecessor'
+      end
+    end
+
+    def mysql_group_count_variable(uid)
+      "@aggregated_journal_row_counter_#{uid}"
+    end
+
+    # Similar to the WHERE statement used in :sql_rough_group. However, this condition will
+    # match (return true) for all pairs where a merge/aggregation IS possible.
+    def sql_on_groups_belong_condition(predecessor, successor)
+      "#{predecessor}.group_number + 1 = #{successor}.group_number AND
+      (NOT #{sql_beyond_aggregation_time?(predecessor, successor)} AND
+      #{predecessor}.user_id = #{successor}.user_id AND
+      #{successor}.journable_type = #{predecessor}.journable_type AND
+      #{successor}.journable_id = #{predecessor}.journable_id AND
+      NOT ((#{predecessor}.notes != '' AND #{predecessor}.notes IS NOT NULL) AND
+      (#{successor}.notes != '' AND #{successor}.notes IS NOT NULL)))"
+    end
+
+    # Returns a SQL condition that will determine whether two entries are too far apart (temporal)
+    # to be considered for aggregation. This takes the current instance settings for temporal
+    # proximity into account.
+    def sql_beyond_aggregation_time?(predecessor, successor)
```

Comment:
This method name should technically probably not end with ? since you are not returning a boolean _value_ but rather a boolean _condition_. Though I am totally fine with how it is.